### ISSUE TITLE
[Q] Was the reported FPS achieved using GPU?

### ISSUE THREAD

**Issue by ValYouW** [navervision/mlsd#29]
Running on Windows using xnnpack delegate I get ~300ms per inference, using model `M-LSD_512_tiny_fp16.tflite`
(relatively old cpu: `Intel(R) Core(TM) i7-2600K CPU @ 3.40GHz, 3401 Mhz, 4 Core(s), 8 Logical Processor(s)`)

For comparison, running object detection using ssd mobilenet v3 (input size 320x320) I get ~17ms per inference

**Comment by geonm** [comment]
Yes. We used V100 GPUs to get all reported FPS numbers with a native ckpt model **not tflite model**.

If you use tflite converted models (we call it tflite model here), it will use only CPU and it sometimes shows slower inference time than mobile devices even it’s on a high-performance CPU (Xeon something) because TFLITE OP Kernels are optimized for ARM processors.

We highly recommend to use tflite models for mobile devices.

There are also the native ckpt models in this repository.

**Comment by ValYouW** [comment]
Hi,

Yes, I am using your provided tflite model (thx!), I'll surely test it on my Android.

As for the SSD, yes, I converted it to tflite, the project I am using is [this](https://github.com/ValYouW/tflite-win-c), you can find there the tflite model.
If I remember correctly I took the model from [here](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/tf2_detection_zoo.md)

Thanks again.

**Comment by ValYouW** [comment]
@geonm 
So I tested on Android, using the `M-LSD_320_tiny_fp16` model, on my Pixel 4A 5G I get ~250ms per detection.
On the same device using [ssd_mobilenet_v1](https://tfhub.dev/tensorflow/lite-model/ssd_mobilenet_v1/1/metadata/2) I get ~50ms per detection.

If you're interested you can run the code from my repo: https://github.com/ValYouW/tflite-crossplatform

Thx.

P.S.
I also tried using the nnapi delegate but there was no real difference...

**Comment by geonm** [comment]
Thank you for reporting valuable numbers.

Honestly, our mlsd model has to be slower than ssd_mobilenet_v1.

We adopt the slightly modified mobilenet as a backbone but we designed a decoder part like U-Net which requires more computational cost than SSD and also we employed a larger input image (320x320)

Unlike object detection, the evaluation metric of line segment detection (especially wireframe dataset) mostly depends on how well a model could detect adjacent line segments.

If we adopt SSD as a decoder, the performance would drastically drop.
